### PR TITLE
Added tests from unusedTestData

### PR DESCRIPTION
**Pull request by mvicsokolova**

### CODE REVIEW

File: src/test/java/kotlinx/reflect/lite/testData/modifiers/functions.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    val externalGetter = clazz.getMemberByName("externalGetter") as KProperty0<Unit>
+    // TODO should be true
+    //assertTrue(externalGetter.getter.isExternal)
+    assertFalse(externalGetter.getter.isInline)
+
+    val inlineProperty = clazz.getMemberByName("inlineProperty") as KMutableProperty0<Unit>
+    assertFalse(inlineProperty.getter.isExternal)
+    // TODO should be true
+    //assertTrue(inlineProperty.getter.isInline)
```
Comment (mvicsokolova) [inline src/test/java/kotlinx/reflect/lite/testData/modifiers/functions.kt:72]: This is delegated to `Flag.Function.IS_INLINE(property.kmProperty.getterFlags)` but it returns false, although the property `inlineProperty`  is inline. I guess, this is an issue in kotlinx.metadata 🙄

File: src/test/java/kotlinx/reflect/lite/testData/modifiers/functions.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    val suspend = Suspend::class.java.kotlin.getMemberByName("suspend") as KFunction<Unit>
+    assertFalse(suspend.isInline)
+    assertFalse(suspend.isExternal)
+    assertFalse(suspend.isOperator)
+    assertFalse(suspend.isInfix)
+    assertTrue(suspend.isSuspend)
+
+    val externalGetter = clazz.getMemberByName("externalGetter") as KProperty0<Unit>
+    // TODO should be true
+    //assertTrue(externalGetter.getter.isExternal)
```
Comment (mvicsokolova) [inline src/test/java/kotlinx/reflect/lite/testData/modifiers/functions.kt]: And here is the same issue: `Flag.Function.IS_EXTERNAL(property.kmProperty.getterFlags)` should be true, but it's not

File: src/test/java/kotlinx/reflect/lite/testData/modifiers/functions.kt (hunk 1 above)
Comment (qwwdfsad) [reply src/test/java/kotlinx/reflect/lite/testData/modifiers/functions.kt:72]: Please file it in YT with self-contained reproducer (without reflect-lite) and post issue ID here in code comment

File: src/test/java/kotlinx/reflect/lite/testData/modifiers/functions.kt (hunk 2 above)
Comment (qwwdfsad) [reply src/test/java/kotlinx/reflect/lite/testData/modifiers/functions.kt]: Same here

File: src/test/java/kotlinx/reflect/lite/testData/modifiers/functions.kt (hunk 1 above)
Comment (mvicsokolova) [reply src/test/java/kotlinx/reflect/lite/testData/modifiers/functions.kt:72]: I've filed the issues ([KT-53862](https://youtrack.jetbrains.com/issue/KT-53862/kotlinxmetadata-jvm-FlagFunctionISINLINEpropgetterFlags-returns-false-for-inline-properties), [KT-53863](https://youtrack.jetbrains.com/issue/KT-53863/kotlinxmetadata-jvm-FlagFunctionISEXTERNALpropgetterFlags-returns-false-for-a-property-with-an-external-getter)) and updated the kotlinx.metadata version to 0.5.0

Comment (qwwdfsad) [reply src/test/java/kotlinx/reflect/lite/testData/modifiers/functions.kt:72]: 👍

**Review by qwwdfsad — APPROVED** [review]
Good to go

**Pull request merged**